Automattic/wp-calypso
# Theme Sheet: Turn ThemeDetails data component into a query component

This is mostly motivated as a preparation for #8133.

This PR also:
* Makes the new query component an ES6 class component.
* Introduces a new `isRequesting` type of selector used by that 

To test:
* Make sure that theme sheets still work:
* Land on calypso.localhost:3000/design, click on any theme
* Land on that theme directly (e.g by reloading when already on a theme sheet page). This is especially relevant when logged out to make sure SSR works (try with JS disabled for good measure).

cc @ehg since it's actually prep work for the smoother single-tree rendering API (#8133)
